src/main/java/com/redhat/red/build/koji/KojiClient.java

```diff
@@ -808,10 +808,64 @@ public KojiWinArchiveInfo getWinArchiveInfo( int archiveId, KojiSessionInfo sess
                                                                STANDARD_REQUEST_MODIFIER );
 
             List<KojiArchiveInfo> archives = response.getArchives();
+
+            if ( archives != null && query.getType() == null )
+            {
+                return updateArchiveTypeInfo( archives, session );
```

Comment:
This is probably wrong. It is trying to do the same thing as with the build type info and just retrieve the missing fields. However, I don't think that it's always the case that the build type is the same as the archive type. That's the case most of the time, but, for example, rpm builds can produce maven artifacts.

The other way to do it is to call `getArchive()` for each archive and get every field again, but this call includes the additional type-specific fields. Unfortunately, these calls are expensive, but  `listArchives()` does not seem to return the archive type on its own. At least it may be possible to do the `getArchive()` call as a multicall.

Reply:
Can we ask that the additional information be made available in the listArchives call, then? It seems like we should hold off on implementing something and setting an expectation with the user that we can't completely support. I'm definitely -1 on iterating + calling getArchive...but I wonder if there's a way to communicate to the user that they will need to do something like a multicall in order to populate this information?

IMO we need to avoid magic and things that have known counter-examples where the user might trip into an unsupported situation with no warning.

Reply:
Actually, the type for builds seems to be a list, or the union of the build types of the archives. This means that each archive only has a single build type. This would mean that an archives `bType` would be a subset of the build's types.

Reply:
> Can we ask that the additional information be made available in the listArchives call, then? It seems like we should hold off on implementing something and setting an expectation with the 

@jdcasey Unfortunately, it does not seem like you can request that it populate the information in the `listArchives()` API call. For this, you need to tell it `type` in the query beforehand, so it will only fill in the information for one given type even, though a `listArchives()` call on its own may return multiple types. This is what creates the issue to begin with.

> user that we can't completely support. I'm definitely -1 on iterating + calling getArchive...but I wonder if there's a way to communicate to the user that they will need to do something like a multicall in order to populate this information?

It's easier to create a multicall if we just multicall a single method like `getArchive()` for the ids returned by `listArchives()`. Or, we could split into 3 multicalls, one for each type. It's hard to know if that preprocessing would be worth it.

> 
> IMO we need to avoid magic and things that have known counter-examples where the user might trip into an unsupported situation with no warning.

What is the counter-example? This same method is being used to get the build type information for builds already. I think the client should support a multicall version of `getBuild()` in which case it can use multicall for the type information as well.

Alternatively, we could provide a separate method in the client to fill in the information, or a flag to an existing method.